### PR TITLE
Authorize websocket requests with protocol header

### PR DESCRIPTION
**Pull request by liggitt**
Requires https://github.com/openshift/origin/pull/14716

### CODE REVIEW

File: dist/origin-web-common-services.js [hunk 1]
```diff
@@ -693,8 +693,23 @@ angular.module('openshiftCommonServices')
...
+          config.protocols = config.protocols || [];
+
+          // Ensure protocols is an array
+          if (!_.isArray(config.protocols)) {
+            config.protocols = [config.protocols];
+          }
+
+          // Ensure protocols has at least one item in it (for the server to echo back once the bearer protocol is stripped out)
+          if (config.protocols.length == 0) {
+            config.protocols.unshift("undefined");
```
Comment (jwforres) [inline dist/origin-web-common-services.js]: could we do something like console.openshift.io.placeholder or something?  I worry that seeing "undefined" in a request handshake might cause someone to open a bug/ticket

**Comment by jwforres** [comment]
Marked as WIP since this depends on another PR that isn't merged

File: dist/origin-web-common-services.js (hunk 1 above)
Comment (liggitt) [inline dist/origin-web-common-services.js]: firefox sends "undefined" today, so I know it works. I guess I don't mind sending a specific placeholder protocol, though it's hard to pick one that I could see upstreaming to kube (which I want to)

**Comment by liggitt** [comment]
@jwforres prereq and upstream are merged, this is ready

**Comment by liggitt** [comment]
Do I need to cut/bump this in the web console repo?

**Comment by jwforres** [comment]
Yeah someone will need to do that

**Comment by jeff-phillips-18** [comment]
There is another change ( #119 ) that requires an update in both origin-web-catalog and origin-web-console before we add to the web console repo.

File: dist/origin-web-common-services.js (hunk 1 above)
Comment (xingxingxia) [inline dist/origin-web-common-services.js]: I'm one of the "worrying" :)
I'm intending to leave comment in card to @liggitt whether "undefined" is indicating something exceptional, and finding you already discussed it. Solved my worrying :) thx

**Pull request merged**